cmake/dependencies/pybind11.cmake

```diff
@@ -37,7 +37,7 @@ function(find_pybind11)
             mark_as_advanced(FETCHCONTENT_UPDATES_DISCONNECTED_FETCHEDpybind11)
         endif()
     else()
-        find_package(pybind11 2.11.1 CONFIG REQUIRED)
+        find_package(pybind11 2.12.0 CONFIG REQUIRED)
```

Comment:
Let's do this next release, I have not updated pyAMReX' pybind11 yet:
https://github.com/AMReX-Codes/pyamrex/blob/24.04/cmake/dependencies/pybind11.cmake

Reply:
https://github.com/AMReX-Codes/pyamrex/pull/291